project-addons/custom_account/models/account.py

```diff
@@ -219,7 +220,7 @@ def invoice_validate(self):
         res = super().invoice_validate()
         for inv in self:
             for line in inv.invoice_line_ids:
-                line.write({'cost_unit': line.product_id.standard_price_2})
+                line.write({'cost_unit': (numpy.average(line.move_line_ids.mapped('price_unit')) * -1) or line.product_id.standard_price_2})
```

Comment:
ponle antes de nada un "line.move_line_ids and" por si a caso sino me imagino que el numpy fallará si recibe None

Reply:
Corregido